internal/cmd/auth.go

```diff
@@ -289,7 +298,25 @@ func logout(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func whoAmI(cmd *cobra.Command, _ []string) error {
+	cmd.SilenceUsage = true
+	client, err := createTursoClientFromAccessToken(false)
+	if err != nil {
+		return err
+	}
+	user, err := client.Users.GetUser()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%s\n", user.Username)
+	return nil
+}
+
 func verifyIfTokenIsSetInEnv(cmd *cobra.Command, args []string) error {
+	// we want to override this for `turso auth whoami`
+	if cmd.Use == whoAmICmd.Use {
+		return nil
+	}
```

Comment:
We have a top level check for `turso auth` commands which prevents usage of `auth` command if `TURSO_API_TOKEN` is set. Just for the `whoami` sub command I have added this exception.